### ISSUE TITLE
megfile.smart_glob missing s3 profile

### ISSUE THREAD

**Issue by NashMiao** [megvii-research/megfile#355]
when I use megfile.smart_glob with profile:

```
megfile.smart_glob('s3+profile://data/*')
```

megfile/s3_path.py:463 will access wrong bucket for `top_dir` not include profile

**Comment by LoveEatCandy** [comment]
Have you setted up s3 configuration of `profile` ? Document about it: https://megvii-research.github.io/megfile/configuration.html#config-for-different-s3-server-or-authentications

**Comment by NashMiao** [comment]
yes, maybe need to correct code like this:

```
        def create_generator(_s3_pathname) -> Iterator[FileEntry]:
            if profile_name:
                correct_top_dir = top_dir.replace('s3://', f's3+{profile_name}://')
            else:
                correct_top_dir = top_dir
            if not S3Path(correct_top_dir).exists():
                return
```

**Comment by LoveEatCandy** [comment]
Sorry, it's a bug. Now fixed in `3.0.2.post1`. You can upgrade and try again. Thank you for your bug report.

**Comment by NashMiao** [comment]
@LoveEatCandy Thank you very much! I've updated to version 3.0.2.post1, and the issue has been resolved.
